base/server/src/main/java/com/netscape/cms/servlet/csadmin/GetConfigEntries.java

```diff
@@ -218,4 +226,28 @@ private String getReplicationPassword() throws EBaseException {
         return pwdStore.getPassword("replicationdb", 0);
     }
 
+    /**
+     * @author Marco Fargetta
+     * @deprecated <subsystem_name>.<cert_id>,certreq configuration properties will be removed in future versions..
+     */
+    @Deprecated (since = "11.5.0")
+    private String getCSR(String param) throws EBaseException {
+        CMSEngine engine = getCMSEngine();
+        EngineConfig config = engine.getConfig();
+        String csr = null;
+
+        String nickname = config.getString(param.replace(".certreq", ".nickname"), null);
+        if (nickname == null || nickname.isEmpty()) {
+            return null;
+        }
+        Path csrConfCertsPath = FileSystems.getDefault().getPath(CMS.getInstanceDir(), "conf", "certs", nickname + ".csr");
+        try {
+            List<String> csrLines = Files.readAllLines(csrConfCertsPath, StandardCharsets.UTF_8);
+            csr = csrLines.stream().filter(x -> !x.startsWith("---")).collect(Collectors.joining());
```

Comment:
This will work but we also have a `CertUtil.unwrapPKCS10()` that validates the header/footer.

Reply:
OK. I am updating with the utility method.